aten/src/ATen/native/GridSampler.cpp

```diff
@@ -380,12 +381,278 @@ namespace {
+          for (int64_t w = 0; w < out_W; ++w) {
+            // get the corresponding input x, y, z co-ordinates from grid
+            scalar_t *grid_ptr_NDHW = grid_ptr_N + h * grid_sH + w * grid_sW;
+            scalar_t ix = *grid_ptr_NDHW;
+            scalar_t iy = grid_ptr_NDHW[grid_sCoor];
+
+            ix = grid_sampler_compute_source_index(ix, inp_W, padding_mode, align_corners);
+            iy = grid_sampler_compute_source_index(iy, inp_H, padding_mode, align_corners);
+
+            if (interpolation_mode == GridSamplerInterpolation::Bilinear) {
+              // get corner pixel values from (x, y, z)
+              // for 4d, we used north-east-south-west
+              int64_t ix_nw = static_cast<int64_t>(std::floor(ix));
+              int64_t iy_nw = static_cast<int64_t>(std::floor(iy));
+
+              int64_t ix_ne = ix_nw + 1;
+              int64_t iy_ne = iy_nw;
+
+              int64_t ix_sw = ix_nw;
+              int64_t iy_sw = iy_nw + 1;
+
+              int64_t ix_se = ix_nw + 1;
+              int64_t iy_se = iy_nw + 1;
+
+
+              // get surfaces to each neighbor:
+              scalar_t nw = (ix_se - ix)    * (iy_se - iy);
+              scalar_t ne = (ix    - ix_sw) * (iy_sw - iy);
+              scalar_t sw = (ix_ne - ix)    * (iy    - iy_ne);
+              scalar_t se = (ix    - ix_nw) * (iy    - iy_nw);
+
+              // calculate bilinear weighted pixel value and set output pixel
+              scalar_t *inp_ptr_NC = inp_ptr_N;
+              scalar_t *out_ptr_NCHW = out_ptr + n * out_sN + h * out_sH + w * out_sW;
+              for (int c = 0; c < C; ++c, out_ptr_NCHW += out_sC, inp_ptr_NC += inp_sC) {
+                auto res = static_cast<scalar_t>(0);
+                if (within_bounds_2d(iy_nw, ix_nw, inp_H, inp_W)) {
+                  res += inp_ptr_NC[iy_nw * inp_sH + ix_nw * inp_sW] * nw;
+                }
+                if (within_bounds_2d(iy_ne, ix_ne, inp_H, inp_W)) {
+                  res += inp_ptr_NC[iy_ne * inp_sH + ix_ne * inp_sW] * ne;
+                }
+                if (within_bounds_2d(iy_sw, ix_sw, inp_H, inp_W)) {
+                  res += inp_ptr_NC[iy_sw * inp_sH + ix_sw * inp_sW] * sw;
+                }
+                if (within_bounds_2d(iy_se, ix_se, inp_H, inp_W)) {
+                  res += inp_ptr_NC[iy_se * inp_sH + ix_se * inp_sW] * se;
+                }
+                *out_ptr_NCHW = res;
+              }
+            } else if (interpolation_mode == GridSamplerInterpolation::Nearest) {
+              int64_t ix_nearest = static_cast<int64_t>(std::round(ix));
+              int64_t iy_nearest = static_cast<int64_t>(std::round(iy));
+
+              // assign nearest neighor pixel value to output pixel
+              scalar_t *out_ptr_NCHW = out_ptr + n * out_sN + h * out_sH + w * out_sW;
+              scalar_t *inp_ptr_NC = inp_ptr_N;
+              for (int c = 0; c < C; ++c, out_ptr_NCHW += out_sC, inp_ptr_NC += inp_sC) {
```

Comment:
Should this be `int64_t c`? (I don't expect that C will get that big, but...)

Reply:
Fixed a few other instances of using `int` for `c` now you mention it. It might be overkill but it certainly doesn't hurt.